andela/ah-scorpion
# #159053990 Add Feature For User Following

**What does this PR do?**

Add a user following functionality.

**Description of Task to be completed?**

_Scenario 1:_
Only authenticated users should be able to follow and or unfollow other logged in users.

_Scenario 2:_
I need a view to manage those I follow and be able to see those who follow me.
Write the tests for user following including the edge cases.

**How to test?**

1. Clone the project from the branch ft-user-following-159053990
2. https://github.com/andela/ah-scorpion.git
3. Create a virtual environment called envvirtualenv env -p python3
4. Activate the virtual environment source env/bin/activate
5. Install the dependencies pip install -r requirements.txt
6. Register multiple users.
7. Login a user who will follow and unfollow other users
8. Test the follow user endpoint using **PUT** request on Postman `http://127.0.0.1:8000/api/v1/profiles/<username>/follow`
9. Test the unfollow user endpoint using **DELETE** request on Postman `http://127.0.0.1:8000/api/v1/profiles/<username>/follow`
10. Check the logged in user's followers
`http://127.0.0.1:8000/api/v1/profiles/followers/`
11. Check the logged in user's following
`http://127.0.0.1:8000/api/v1/profiles/following/`
12. Run the tests coverage run --source='.' manage.py test

What are the relevant pivotal tracker stories?

[#159053990](https://www.pivotaltracker.com/story/show/159053990)
